yarn-project/archiver/src/archiver/kv_archiver_store/contract_class_store.ts

```diff
@@ -1,20 +1,20 @@
-import { Fr, FunctionSelector } from '@aztec/circuits.js';
+import { Fr, FunctionSelector, Vector } from '@aztec/circuits.js';
 import { BufferReader, numToUInt8, serializeToBuffer } from '@aztec/foundation/serialize';
 import { AztecKVStore, AztecMap } from '@aztec/kv-store';
-import { ContractClassPublic } from '@aztec/types/contracts';
+import { ContractClassPublic, ExecutablePrivateFunctionWithMembershipProof } from '@aztec/types/contracts';
 
 /**
  * LMDB implementation of the ArchiverDataStore interface.
  */
 export class ContractClassStore {
   #contractClasses: AztecMap<string, Buffer>;
 
-  constructor(db: AztecKVStore) {
+  constructor(private db: AztecKVStore) {
     this.#contractClasses = db.openMap('archiver_contract_classes');
   }
 
-  addContractClass(contractClass: ContractClassPublic): Promise<boolean> {
-    return this.#contractClasses.set(contractClass.id.toString(), serializeContractClassPublic(contractClass));
+  async addContractClass(contractClass: ContractClassPublic): Promise<void> {
```

Comment:
Is it not still worth returning the result?

Reply:
Just changed it to return the result, even though it gets discarded by the callers. Still, I dug a bit deeper to figure out where that boolean is coming from: it's from lmdb's `put` method that [may return false if the insertion doesn't happen due to a versioning setting, which we're not using](https://www.npmjs.com/package/lmdb#:~:text=The%20promise%20will%20resolve%20once%20the%20transaction%20has%20finished%20committing.%20The%20resolved%20value%20of%20the%20promise%20will%20be%20true%20if%20the%20put%20was%20successful%2C%20and%20false%20if%20the%20put%20did%20not%20occur%20due%20to%20the%20ifVersion%20not%20matching%20at%20the%20time%20of%20the%20commit.). So I think we should just update all those booleans to just voids in the `Map` interface. @alexghr thoughts, given you had worked on it extensively?

Reply:
Sounds good to me!